nav2_amcl/src/amcl_node.cpp

```diff
@@ -686,7 +692,6 @@ AmclNode::externalPoseReceived(const geometry_msgs::msg::PoseWithCovarianceStamp
 void
 AmclNode::handleInitialPose(geometry_msgs::msg::PoseWithCovarianceStamped & msg)
 {
-  std::lock_guard<std::recursive_mutex> cfl(mutex_);
   // In case the client sent us a pose estimate in the past, integrate the
```

Comment:
some of the variables referenced below are accessed from different threads, but it looks like these are not perhaps properly protected anyhow.  If you add lock the mutex in `on_configure` I think that will cover everything, does it work?
Do you know what thread is used for `on_configure`?  to be "correct", we'd have to take copies of all the vars that are written in a different thread, with the lock held.  I think its just `base_frame_id`, `odom_frame_id` and `transform_lookup_timeout_`, even though they are effectively only written once